core/get_ratings.py

```diff
@@ -1,148 +1,141 @@
 #!/usr/bin/env python3
 
-from datetime import datetime
+from collections import namedtuple
 import json
 import os
 
 from absl import flags, app
 import numpy as np
 import pandas as pd
 
-from common import date_is_correct
+from common import date_is_correct, today
 import constants
 
+Players = namedtuple('Players', ['winner', 'looser'])
+ScoreInSets = namedtuple('Score', ['winner', 'looser'])
+GameRatings = namedtuple('GameRatings', ['my', 'opponent'])
+
 FLAGS = flags.FLAGS
-script_dir = os.path.dirname(os.path.abspath(__file__))
-repo_root_dir = os.path.join(script_dir, '..')
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+REPO_ROOT_DIR = os.path.join(SCRIPT_DIR, '..')
 
-flags.DEFINE_string('day', None, 'Day of leaderboard in format yyyy-mm-dd.')
+flags.DEFINE_string('d', today(), ('Day of leaderboard to print in format yyyy-mm-dd. '
+                                   'If day is not specified or its value is equal to today(), '
+                                   'script will update json data.'))
 
 
-def win_set_probability(my_rating, opponent_rating):
-    return 1.0 / (1 + 10 ** ((opponent_rating - my_rating) / 400))
+def win_set_probability(game_ratings):
+    return 1.0 / (1 + 10 ** ((game_ratings.opponent - game_ratings.my) / 400))
 
 
 def C(k, n):
     return np.math.factorial(n) / np.math.factorial(k) / np.math.factorial(n - k)
 
 
-def expected_sets_won(my_rating, opponent_rating, n_sets):
-    win_prob = win_set_probability(my_rating, opponent_rating)
+def expected_sets_won(game_ratings, n_sets):
+    win_prob = win_set_probability(game_ratings)
     sets = np.arange(n_sets + 1)
     win_sets_probs = np.array([C(i, n_sets) * (win_prob ** i) * ((1 - win_prob) ** (n_sets - i)) for i in sets])
     return np.sum(win_sets_probs * sets)
 
 
 def get_norm_coeff(old_rating, sets_played):
-    if sets_played < 100:
-        return 40
-    elif old_rating >= 2400:
-        return 10
-    else:
-        return 20
+    return 40 if sets_played < 100 else 10 if old_rating >= 2400 else 20
```

Comment:
maybe extract `100` and `2400` to constants?

Reply:
OK